File: albedo-math/src/test/java/jp/albedo/jpl/files/util/EphemerisSecondsTest.java
package jp.albedo.jpl.files.util;

import jp.albedo.common.JulianDay;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class EphemerisSecondsTest {

    @Test
    public void testFromJde() {
        assertEquals(-3.277299484800115E8, EphemerisSeconds.fromJde(2447751.8293));
        assertEquals(-14200747200.0, EphemerisSeconds.fromJde(JulianDay.fromDate(1549, 12, 31)));
        assertEquals(20514081600.0, EphemerisSeconds.fromJde(JulianDay.fromDate(2650, 1, 25)));
    }

    @Test
    public void testToJde() {
        assertEquals(2447751.8293, EphemerisSeconds.toJde(-3.277299484800115E8));
        assertEquals(JulianDay.fromDate(1549, 12, 31), EphemerisSeconds.toJde(-14200747200.0));
        assertEquals(JulianDay.fromDate(2650, 1, 25), EphemerisSeconds.toJde(20514081600.0));
    }

}

File: api/math/jpl/EphemerisSeconds.test.ts
import { describe, it, expect } from "vitest";
import { JulianDay } from "../";
import { EphemerisSeconds } from './';

describe("EphemerisSeconds", () => {

    it("should calculate from JDE", () => {
        expect(EphemerisSeconds.fromJde(2447751.8293)).toBe(-3.277299484800115E8);
        expect(EphemerisSeconds.fromJde(JulianDay.fromDate(1549, 12, 31))).toBe(-14200747200.0);
        expect(EphemerisSeconds.fromJde(JulianDay.fromDate(2650, 1, 25))).toBe(20514081600.0);
    });

    it("should calculate to JDE", () => {
        expect(EphemerisSeconds.toJde(-3.277299484800115E8)).toBe(2447751.8293);
        expect(EphemerisSeconds.toJde(-14200747200.0)).toBe(JulianDay.fromDate(1549, 12, 31));
        expect(EphemerisSeconds.toJde(20514081600.0)).toBe(JulianDay.fromDate(2650, 1, 25));
    });

});
